### PR TITLE
Is2279/jup lab studies dispatcher (⚠️ devops)

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Studies dispatcher can now dispatch services that consume a file (based on filetype) ONLY if user is **registered**.
- adds extra column to ``services_consume_filetypes``
- adapts tests, minor fixes and doc
- @ignapas can you setup please the portal in your computer and point it to master? Are there any dataset example with files having PY and IPYNB extensions?
 - ⚠️ devops: requires updating table ``services_consume_filetypes`` with new jupyterlab service
```sql
('simcore/services/dynamic/jupyter-octave-python-math',	'1.6.9',	'JupyterLab Math',	'input_1',	'PY',	0,	'0'),
('simcore/services/dynamic/jupyter-octave-python-math',	'1.6.9',	'JupyterLab Math',	'input_1',	'IPYNB',	0,	'0')
```

### Following this PR
- Enhancement in the UI should follow this PR (see #2283)
- New e2e script to check stability of the workflow
- 

## Related issue/s

#2279

## How to test

```cmd
cd osparc-simcore
make devenv
source .venv/bin/acticate
cd services/web/server
make install-dev
make test-dev-unit
```

Or manually:
- build& start the stack
```
cd osparc-simcore
make build-x
make up-prod
```
Click any of the link in https://github.com/pcrespov/osparc-sample-studies/tree/master/files%20samples

### CODE REVIEW

**Review by odeimaiz — APPROVED** [review]

**Review by GitHK — COMMENTED** [review]
One concern: how do we avoid abuse now that guests are allowed?

**Comment by pcrespov** [comment]
> now that guests are allowed?

now? this has always been like that. Actually this PR adds a restriction to that, i.e. some viewers are ONLY accessible to registered users

File: services/web/server/src/simcore_service_webserver/studies_dispatcher/__init__.py [hunk 1]
```diff
@@ -1,6 +1,15 @@
 """
     This app module dispatches pre-configure or on-the-fly studies to user from a link request
     that is redirected to the front-end which customizes its view
+
+    - Table services_consume_filetypes defines a map between a servive (key,version,input_port) and a filetype
+    - Exposes viewers resource in API (handlers_rest.py)
+    - Entrypoint that dispatches a study to download & view a file (handlers_redirects.py)
+        - finds default viewer
+        - get or create user
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/studies_dispatcher/__init__.py:9]: do you create a user there? I thought these were only logged in users?

File: services/web/server/src/simcore_service_webserver/studies_dispatcher/_users.py [hunk 2]
```diff
@@ -145,3 +108,43 @@ async def _create_temporary_user(request: web.Request):
         )
 
     return user
+
+
+async def acquire_user(request: web.Request, *, is_guest_allowed: bool) -> UserInfo:
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/studies_dispatcher/_users.py:113]: ok understood

File: services/web/server/src/simcore_service_webserver/studies_dispatcher/handlers_redirects.py [hunk 3]
```diff
@@ -51,8 +51,8 @@ def create_redirect_response(
 # HANDLERS --------------------------------
 class ViewerQueryParams(BaseModel):
     file_type: str
-    viewer_key: constr(regex=KEY_RE)
-    viewer_version: constr(regex=VERSION_RE)
+    viewer_key: constr(regex=KEY_RE)  # type: ignore
+    viewer_version: constr(regex=VERSION_RE)  # type: ignore
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/studies_dispatcher/handlers_redirects.py:55]: are these also for mypy?

File: services/web/server/src/simcore_service_webserver/studies_dispatcher/handlers_redirects.py [hunk 4]
```diff
@@ -77,9 +77,8 @@ def decode_downloadlink(cls, v):
     @classmethod
     def from_request(cls, request: web.Request) -> "RedirectionQueryParams":
         try:
-            obj = cls(**dict(request.query))
+            obj = cls.parse_obj(dict(request.query))
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/studies_dispatcher/handlers_redirects.py:80]: what about parse_orm ?

**Review by sanderegg — APPROVED** [review]
small things to check. cool!

File: services/web/server/src/simcore_service_webserver/studies_dispatcher/handlers_redirects.py (hunk 3 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/studies_dispatcher/handlers_redirects.py:55]: yes

File: services/web/server/src/simcore_service_webserver/studies_dispatcher/handlers_redirects.py (hunk 4 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/studies_dispatcher/handlers_redirects.py:80]: ``request.query`` returns a ``MultiDict`` and ``.parse_orm`` is of [ORM-like](https://pydantic-docs.helpmanual.io/usage/models/#orm-mode-aka-arbitrary-class-instances) instances, i.e. with dotted attributes 
for instance ``a.foo`` vs ``a["foo"]``. The former is ORM while for the latter one shall parse it with ``.parse_obj``

**Pull request merged**